### PR TITLE
[all] Implement getModuleComponentList for most modules

### PR DESCRIPTION
**Pull request by alxbilger**
______________________________________________________

By submitting this pull request, I acknowledge that  
**I have read, understand, and agree [SOFA Developer Certificate of Origin (DCO)](https://github.com/sofa-framework/sofa/blob/master/CONTRIBUTING.md#sofa-developer-certificate-of-origin-dco)**.
______________________________________________________

**Reviewers will merge this pull-request only if**  
- it builds with SUCCESS for all platforms on the CI.
- it does not generate new warnings.
- it does not generate new unit test failures.
- it does not generate new scene test failures.
- it does not break API compatibility.
- it is more than 1 week old (or has fast-merge label).

### CODE REVIEW

**Comment by alxbilger** [comment]
[ci-build][with-all-tests]

File: Sofa/Component/AnimationLoop/src/sofa/component/animationloop/init.cpp [hunk 1]
```diff
@@ -54,4 +55,10 @@ void init()
     }
 }
 
+const char* getModuleComponentList()
+{
+    /// string containing the names of the classes provided by the plugin
+    static std::string classes = core::ObjectFactory::getInstance()->listClassesFromTarget(sofa_tostring(SOFA_TARGET));
```
Comment (fredroy) [inline Sofa/Component/AnimationLoop/src/sofa/component/animationloop/init.cpp]: I think the constexpr var `MODULE_NAME` (which is defined for each module normally, and in its own namespace) would be better than to use the SOFA_TARGET preproc macro

**Review by fredroy — CHANGES_REQUESTED** [review]

File: Sofa/Component/AnimationLoop/src/sofa/component/animationloop/init.cpp (hunk 1 above)
Comment (alxbilger) [reply Sofa/Component/AnimationLoop/src/sofa/component/animationloop/init.cpp]: you're right. Done

File: Sofa/Component/AnimationLoop/src/sofa/component/animationloop/init.cpp [hunk 2]
```diff
@@ -54,4 +55,10 @@ void init()
     }
 }
 
+const char* getModuleComponentList()
+{
+    /// string containing the names of the classes provided by the plugin
+    static std::string classes = core::ObjectFactory::getInstance()->listClassesFromTarget(sofa_tostring(MODULE_NAME));
```
Comment (fredroy) [inline Sofa/Component/AnimationLoop/src/sofa/component/animationloop/init.cpp]: I dont think `sofa_tostring` is still required ?

**Review by fredroy — CHANGES_REQUESTED** [review]

File: Sofa/Component/AnimationLoop/src/sofa/component/animationloop/init.cpp (hunk 2 above)
Comment (alxbilger) [reply Sofa/Component/AnimationLoop/src/sofa/component/animationloop/init.cpp]: you're right

**Review by fredroy — APPROVED** [review]

**Pull request merged**